### DISCUSSION TITLE
[feat] Rewards for running a full node

### DISCUSSION THREAD

**Discussion by bingoko** [dogecoin/dogecoin#2574] [General]
Could we add a new feature to allow the hoster of the full node can receive a small amount reward to incentivise more and more hoster to join?

**Comment by hzkxf4** [comment]
I would like to see this as well.

**Comment by patricklodder** [comment]
How would you like to fund these rewards?

**Reply by kevstar0** [reply]
The nodes verify the blocks anyway while active.
It's actively mining but it baffles me how nodes dont get rewarded.
It's supposed to be decentralized while we know small devices cant handle an entire blockchain let the nodes take care of that by either generating new coins to our wallets when a blockchain has been fully verified also not forgeting the other variables that come into play but

I strongly believe nodes gotta get some, we are literally helping millions from the bottom of our hearts (proof of work) yet.... why nodes aint getting rewarded for their services (storage, cpu, graphics, ASICS, etc...)

**Reply by patricklodder** [reply]
> It's supposed to be decentralized while we know small devices cant handle an entire blockchain 

You can run your node in pruned mode on a small device. From `dogecoind -help`:

```
  -prune=<n>
       Reduce storage requirements by enabling pruning (deleting) of old
       blocks. This allows the pruneblockchain RPC to be called to
       delete specific blocks, and enables automatic pruning of old
       blocks if a target size in MiB is provided. This mode is
       incompatible with -txindex and -rescan. Warning: Reverting this
       setting requires re-downloading the entire blockchain. (default:
       0 = disable pruning blocks, 1 = allow manual pruning via RPC,
       >2200 = automatically prune block files to stay under the
       specified target size in MiB)
```

> It's actively mining but it baffles me how nodes dont get rewarded.

This is not true. Validating 1 hash is not the same as mining: Mining is the process where you don't know yet what a valid hash will be and you run the scrypt algorithms billions of times, to find a valid hash. Validating a hash that is claimed to be valid, only requires one run of the scrypt algorithm.

**Comment by hzkxf4** [comment]
Thank you for responding. I apologize in advance if my limited understanding of things crypto makes my suggestion idiotic.  Is there much 'dust' in doge transactions?  Could a common dust bin address be created to collect all dust and distribute it to active nodes whenever the accumulated dust reaches a threshold where the bin is full and an even distribution can be made?  I have no idea if this is feasible.  
Thanks again for the response and your work!

**Reply by bingoko** [reply]
Can you give more description on the 'dust' in the transaction?

**Reply by hzkxf4** [reply]
Think Richard Pryor in Superman III or [Office Space](https://www.youtube.com/watch?v=yZjCQ3T5yXo)
Crypto dust is created after a trade occurs, where the trade leaves a trace amount leftover in the address it was sent from. This unspent transaction output (UTXO) is called crypto dust. The asset is basically stuck since it is too small to cover transaction fees or is below the exchange’s minimum trading limit. In turn, the minimum value at which an asset is usable is called the “dust limit.”
source: https://phemex.com/academy/what-is-crypto-dust

**Reply by bingoko** [reply]
I see. Very interesting point.
Then this raises another problem, even it is dust, it is still a personal asset.
Only the sender has the right to donate it to the community.

**Reply by 0foxes** [reply]
From what I understand, we could fund this from the block rewards, transaction fees (including dust which has been discussed), or between the participants (all nodes put up dogecoin, the network moves dogecoin of less reliable nodes to more reliable nodes?). Any of them would require a lot of work though.

**Reply by 0foxes** [reply]
Alternatively, there could be an official or unofficial initiative to collect donations, monitor node activity, and distribute rewards to the nodes on a regular basis. Do you think that would pick up enough traction?

**Reply by bingoko** [reply]
I think 'official initiative to collect donations' is the easiest way to do that at the moment to attract more traction.
Then, splitting the block rewards, transaction fees are easier than collecting the dust. 
It could be the next step (next minor version).

**Reply by Chaudhrycool** [reply]
#2163

**Reply by qlpqlp** [reply]
What about, on newer version, wen installing/upgrading a node simple prompot the question if the user wants to donate, by default, all dust to nodes that have validated the user transactions?

**Reply by kitcateleven** [reply]
If we donate the dust to a bin, when we upgrade, could we use that dust to enable new communities to use crypto for solving hunger/poverty issues? If I didn't get here the exact same way (bought 50k shards of doge coin this time last year) I'd make fun of myself for "pipe dreams" .  But, I did. The dogecoin shards lifted me out of poverty - it has to be scalable.

**Reply by patricklodder** [reply]
> If we donate the dust to a bin, when we upgrade, could we use that dust to enable new communities to use crypto for solving hunger/poverty issues? 

Could you elaborate further? Do you mean to have the wallet do some form of charity on dust / change?

**Reply by kitcateleven** [reply]
I think the network could aggregate the change, from each transaction, beyond a certain decimal point. Then the foundation could manage how it is spent. Also, the network could program the wallets to return dust to the network (for future use). I apologize because I don't have a full understanding of the programming required to do that (though I'm eager to learn). It seems like this dust could be an investment in the future of the network and the ideas behind the manifesto (giving us room to "do only good every day" in the future)

**Reply by kitcateleven** [reply]
For example - If we say the decimal point is 1.00, and a transaction costs 1 doge plus .02 doge for fees, then the purchaser gets charged 1 doge for the transaction (.02doge (fee) + .98doge (rounding the transaction up to 1.00). The network pays the miner and keeps the change. For the priviliege of using doge, the wallet holder pays 1 doge, or less, per transaction while helping the network build reserves for other goals.

**Reply by patricklodder** [reply]
If you want an organization to take a fee out of a transaction, then this should imho be done on a for-purpose wallet, not this one. There's nothing standing in the way of a charity - or otherwise non-profit org - to make their own wallet and implement your scheme.

I don't think that on a protocol level, this should be enforced though. Doing good every day can take many forms, but automatically taxing shibes to force it upon every participant sets a very dangerous precedent and should in my opinion not be done - it is extremely likely to get abused.

**Reply by kitcateleven** [reply]
That makes sense Patrick. I agree

**Comment by kevstar0** [comment]
Great insight but still getting this far as a self-taught web developer we
better recognize our scarcity into going the extreme lengths just to
maintain a node and actually get rewarded 10k doges

the reality of life is we use power to run these machines and not all of us
may or may not be paying for it truth is an effort was made

now how a node gets rewarded for being active will be determined by
verification of blocks and what not and such process renders the computer
useless due to its resources being devoted to finding the next hash

On Tue, Mar 22, 2022 at 11:43 PM OnlyDogecoin ***@***.***>
wrote:

> I also apologize, thanks for your time. But I'm very interested by this
> feature request as I also thought about it few months ago (during the
> Twitter Doge nodes promotion effort :)) ...
>
> So I was also thinking that such feature would allow Dogecoin to promote
> the install/running of its Core node by common (i.e: non technicaly
> interested) people.
>
> By paying Core node owners a random reward/tips, it would be a way to
> promote the expansion of the Dogecoin core nodes network without too much
> human effort. I think we all agree that even something ridiculous like a
> dust incentive (e.g: 1*10^8 DOGE) would help to attract people to install
> it (better than nothing!).
>
> I was understanding that the more there are dogecoin nodes, the more the
> network is resilient, protected and fast.
>
> In fact tbh, I have started to develop a web project to realize such
> objective, but I'm a slow and I stopped dev more than one month ago because
> I wasn't 100% convinced that my way was the good way ...
>
> Because :
> 1 - I had to discover all nodes owners IP and maybe it was a bit dirty
> (network consuming) to peer node connect/disconnect to scan then entire
> network just to discover it (No?).
> 2 - The owner had to claim his node (with his IP) and provide his public
> key to claim his rewards. Because I cannot scan the public key of peer
> nodes (Right ? except if it is in the node subversion ;))
> 3 - The system was relying on community donation to reward random/each
> claimed nodes.
>
> If such feature shouldn't be developped within the Core node project. What
> do you think about my web project ? Do you think it worth to go on this web
> project ? Do you see a better way to discover nodes and collect owners
> public key ?
>
> Thanks and have a good evening!
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/dogecoin/dogecoin/discussions/2574#discussioncomment-2418100>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AJDCPGCLJ4MGCSLR5JQINPTVBI5JHANCNFSM5EC7WXCQ>
> .
> You are receiving this because you commented.Message ID:
> ***@***.***>
>


-- 
Be happy

**Comment by ghost** [comment]
I think that we should create a token on the dogecoin blockchain. This token would have a large but limited supply, but would always be equal to 1 DOGE. This token would be given to nodes for uptime and service. This token would be good for the dogecoin economy because the price of the token could go up with rarity, and therefore would the price of dogecoin. I say 90% of the tokens are locked up for nodes, and 10% goes to hodlers, to build interest.

**Reply by patricklodder** [reply]
If you want to ensure a 1:1 peg, it means you need to be able to offer a 1:1 buyback somewhere. That would mean issuing the same amount of DOGE as you would this token, which then would:

a) Increase DOGE supply.
b) Eradicate the use case of said token, because you might as well issue DOGE.

**Comment by oldcityws** [comment]
Moving to a proof of stake model would lessen the environmental impact, and it can be cheaper / more equitable.  
Not everyone can afford equipment / electricity to mine.  A lot more people can afford to run a core node.
